lib/chai/core/assertions.js

```diff
@@ -537,8 +537,8 @@ module.exports = function (chai, _) {
         itemsCount = val.length;
         break;
       case 'function':
-        var name = val.name ? ' ' + val.name : '';
-        throw new TypeError('.empty was passed a function' + name);
+        var msg = '.empty was passed a function ' + _.getName(val);
+        throw new TypeError(msg.trim());
```

Comment:
The `trim` in this case isn't being used because `getName` might return a string with spaces on the end (as far as I can tell it wouldn't) but rather because if `getName` returned an empty string then the `msg` that's being constructed on line 540 would have a space on the end. In other words, the `trim` would still be needed here even if `getName` did it itself.